aiida/manage/configuration/config.py

```diff
@@ -545,42 +545,20 @@ def remove_profile(self, name):
         self._profiles.pop(name)
         return self
 
-    def delete_profile(
-        self,
-        name: str,
-        include_database: bool = True,
-        include_database_user: bool = False,
-        include_repository: bool = True
-    ):
+    def delete_profile(self, name: str, delete_storage: bool = True) -> None:
         """Delete a profile including its storage.
 
-        :param include_database: also delete the database configured for the profile.
-        :param include_database_user: also delete the database user configured for the profile.
-        :param include_repository: also delete the repository configured for the profile.
+        :param delete_storage: Whether to delete the storage with all its data or not.
         """
-        import shutil
-
-        from aiida.manage.external.postgres import Postgres
+        from aiida.plugins import StorageFactory
 
         profile = self.get_profile(name)
 
-        if include_repository:
-            # Note, this is currently being hardcoded, but really this `delete_profile` should get the storage backend
-            # for the given profile and call `StorageBackend.erase` method. Currently this is leaking details
-            # of the implementation of the PsqlDosBackend into a generic function which would fail for profiles that
-            # use a different storage backend implementation.
-            from aiida.storage.psql_dos.backend import get_filepath_container
-            folder = get_filepath_container(profile).parent
-            if folder.exists():
-                shutil.rmtree(folder)
-
-        if include_database:
-            postgres = Postgres.from_profile(profile)
-            if postgres.db_exists(profile.storage_config['database_name']):
-                postgres.drop_db(profile.storage_config['database_name'])
-
-        if include_database_user and postgres.dbuser_exists(profile.storage_config['database_username']):
-            postgres.drop_dbuser(profile.storage_config['database_username'])
+        if delete_storage:
+            print('STORAGE CONFIG', profile.storage_config)
```

Comment:
Ah, maybe one more nitpick: Do we want to keep this `print` here? Or maybe introduce a nice `echo_report` in the `verdi profile delete` command with more information? With three cases:

1. `psql_dos`: "Dropped Postgres database `<DATABASE_NAME>` of user `<DATABASE_USERNAME>` and removed repository at `<REPOSITORY_URI>`." (maybe also note that he user still exists in case we don't offer this option.)
2. `sqlite_dos`: "Removed SQLite storage at `<FILEPATH>`."
3. `sqlite_zip`: "Removed archive at `<FILEPATH>`."

Reply:
we definitely don't want to keep the print, that was a mistake. We could indeed add a report log message with more details.